experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,573 @@
+        end
+    end
+    
+    #get corresponding face defining it by a system of inequalities
+    F_poly = []
+    m = length(gens(grading_group(R_Q)))
+    C = convex_hull(zeros(Int, m))
+    A = Nothing
+    b = Nothing
+    for a in L_F[2]
+        A,b = compute_inequalities(a)
+        C_a = polyhedron(A,b)
+        C = C + C_a
+        push!(F_poly,[C,A,b])
+    end
+    if L_F[2] == Any[]
+        A = zeros(Int, m,m)
+        b = zeros(Int, m)
+        push!(F_poly,[C,A,b])
+    end
+    return F_poly
+end
+
+##get all prime ideals P_F associated to M
+##INPUT:    ideal I
+##OUTPUT:   list of all minimal associated primes + the maximal ideal and the corresponding face F as a polyhedron  
+function get_all_ass_primes(I)
+    R_Q = base_ring(I)
+    I_m = ideal(R_Q,gens(R_Q)) #maximal ideal
+    F_m = get_face_to_prime(I_m)[1]
+    P_F = [FaceQ(I_m,F_m[1],F_m[2],F_m[3])]
+    MP = minimal_primes(I)
+    for mp in MP
+        p_f = get_face_to_prime(mp)[1]
+        push!(P_F,FaceQ(mp,p_f[1],p_f[2],p_f[3]))
+    end
+    return P_F
+end
+
+
+##Let W = k{Q\(a + F - Q)} and let H be a hyperplane containing F as a subset.
+##Calculates generators of k{(a + H_+^°)\cap Q}.
+##INPUT:    semigroup Q_P as a polyhedron
+#           zonotope G_Q of Q_P as in Lemma 3.10
+#           list of facets Fs of Q_P as polyhedron 
+#           hyperplane h bounding Q as FaceQ
+#           a \in \ZZ^d
+##OUTPUT:   finite set B such that (x^b : b \in B) equals k{(a + H_+^°)\cap Q}
+function generators_W_H(Q_P::Polyhedron,G::Polyhedron,Fs::Vector{Polyhedron{QQFieldElem}},H::FaceQ,a::Vector{Int})
+    F = intersect(H.poly,Q_P)
+
+    #get faces of Q intersecting F only at 0 in Q
+    D = []
+    for f in Fs
+        if dim(intersect(f,F)) == 0
+            push!(D,f)
+        end
+    end
+
+    B = []
+    PaF = polyhedron(H.A,H.A*a+H.b) #a + RR h
+    for d in D
+        I = intersect(PaF,d) #(a + RR H)\cap RR_+D
+        if dim(I) >= 0
+            push!(B,lattice_points(I + G))
+        end
+    end
+
+    #eliminate lattice points on (a + \RR H)
+    B_red = []
+    if length(B) > 0
+        for b in B[1]
+            if dim(intersect(convex_hull(b),PaF)) < 0
+                push!(B_red,b)
+            end
+        end
+    end
+    return B_red
+end
+
+@doc raw"
+    compute_bass_numbers(I,i)
+
+    computes degrees of non-zero Bass numbers up to some given cohomological degree
+
+    INPUT:  monomial ideal I
+            integer i
+    OUTPUT: list of ZZ^d graded degrees of Bass numbers up to cohomological degree i
+"
+function compute_bass_numbers(I,i)
+    R_Q = base_ring(I)
+    I_R = modulus(R_Q)
+    if is_zero(I_R) #R_Q polynomial ring
+        S = R_Q
+    else # R_Q quotient ring
+        S = base_ring(R_Q)
+    end
+
+    # residue_field
+    I_m = ideal(S,gens(S))
+    R_k,_ = quotient_by_ideal(I_m)
+
+    # module R_Q/I as a subquotient of S^1
+    M_mod = quotient_ring_as_module(quo(R_Q,I)[1])
+    
+    D = []
```

Comment:
That's a `Vector{Any}`. Don't you know a bit more about the return value? Then you should use that here, e.g. for a `Vector{Vector{Int}}` you'd write
```suggestion
    D = Vector{Int}[]
```